clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -117,14 +115,15 @@
  * as well as a background I/O thread that is responsible for turning these records into requests and transmitting them
  * to the cluster. Failure to close the producer after use will leak these resources.
  * <p>
- * The {@link #send(ProducerRecord) send()} method is asynchronous. When called it adds the record to a buffer of pending record sends
+ * The {@link #send(ProducerRecord) send()} method is asynchronous. When called, it adds the record to a buffer of pending sending record
  * and immediately returns. This allows the producer to batch together individual records for efficiency.
  * <p>
  * The <code>acks</code> config controls the criteria under which requests are considered complete. The "all" setting
  * we have specified will result in blocking on the full commit of the record, the slowest but most durable setting.
  * <p>
- * If the request fails, the producer can automatically retry, though since we have specified <code>retries</code>
- * as 0 it won't. Enabling retries also opens up the possibility of duplicates (see the documentation on
+ * If the request fails, the producer can automatically retry. We have specified <code>retries</code> default as Integer.MAX_VALUE.
+ * It's recommended to use <code>delivery.timeout.ms</code> to control retry behavior, instead of <code>retries</code>.
+ * Enabling retries also opens up the possibility of duplicates (see the documentation on
```

Comment:
After [KIP-91](https://cwiki.apache.org/confluence/display/KAFKA/KIP-91+Provide+Intuitive+User+Timeouts+in+The+Producer), we set `retries` as max value, and it's recommended to use `delivery.timeout.ms` to control retry behavior. Update it.